eng/tools/eslint-plugin-tsv/src/utils/rule-creator.ts

```diff
@@ -0,0 +1,128 @@
+import { Rule } from "eslint";
+import { TypeSpecConfig } from "../config/types.js";
+import { CreateCodeGenSDKRuleArgs, KeyType, RuleDocument, RuleInfo } from "../interfaces/rule-interfaces.js";
+import { defaultMessageId, defaultRuleType, emitters } from "./constants.js";
+import { NamedRule } from "../interfaces/named-eslint.js";
+import { AST, getStaticYAMLValue } from "yaml-eslint-parser";
+import { createRuleDocument } from "./rule-doc.js";
+
+export function createRule(ruleContext: RuleInfo): NamedRule.RuleModule {
+  const rule: NamedRule.RuleModule = {
+    name: ruleContext.name,
+    meta: {
+      type: defaultRuleType,
+      docs: {
+        description: ruleContext.documentation.description,
+      },
+      schema: [],
+      messages: ruleContext.functions.messages(),
+    },
+    create(context) {
+      return {
+        YAMLDocument(node: Rule.Node) {
+          const yamlDocument = node as unknown as AST.YAMLDocument;
+          const rawConfig = getStaticYAMLValue(yamlDocument) || {};
+          const config = rawConfig as unknown as TypeSpecConfig;
+
+          if (!ruleContext.functions.condition(config, context)) return;
+          ruleContext.functions.validation(config, context, node);
+        },
+      };
+    },
+  };
+  return rule;
+}
+
+export function createRuleMessages(messageId: string, docs: RuleDocument) {
+  return {
+    [messageId]: `${docs.error}.\n${docs.action}.\n${docs.example}`,
+  };
+}
+
+export function isAzureSDK(tspconfig: TypeSpecConfig, emitterName: string) {
+  const flavor = tspconfig.options?.[emitterName]?.flavor as string;
+  return flavor === "azure";
+}
```

Comment:
We may no longer use flavor=azure in emitter option.

As the emitter for branded and unbranded is already different npm package, the emitter used here is always for Azure alone.